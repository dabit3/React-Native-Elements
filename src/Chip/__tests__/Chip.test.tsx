import React from 'react';
import { Chip } from '../index';
import { renderWithTheme } from '../../../.ci/testHelper';
import { fireEvent } from '@testing-library/react-native';

describe('Chip Component', () => {
  it('should render without issues', () => {
    const { getByA11yRole, toJSON } = renderWithTheme(<Chip />);
    const component = getByA11yRole('button');
    expect(component.props.style.borderRadius).toBe(30);
    expect(toJSON()).toMatchSnapshot();
  });

  // Test for solid and outline type
  it.each`
    type
    ${'solid'}
    ${'outline'}
  `('should render $type', ({ type }) => {
    const { queryByText } = renderWithTheme(<Chip title={type} type={type} />);
    expect(queryByText(type)).not.toBeNull();
  });

  it('should pass the onPress function when specified', () => {
    const handlePress = jest.fn();
    const { getByA11yRole } = renderWithTheme(<Chip onPress={handlePress} />);
    fireEvent(getByA11yRole('button'), 'press');
    expect(handlePress).toHaveBeenCalledTimes(1);
  });
});
