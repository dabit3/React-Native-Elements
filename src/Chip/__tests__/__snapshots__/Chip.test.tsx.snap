// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      undefined,
      false,
    ]
  }
  testID="RNE_BUTTON_WRAPPER"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderColor": "#2089dc",
        "borderRadius": 30,
        "borderWidth": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  />
</View>
`;
